### ISSUE TITLE
cffi.FFIError when gmp.dll available on Windows

### ISSUE THREAD

**Issue by twangboy** [Legrandin/pycryptodome#427]
I'm trying to fix an issue we're seeing with Salt (https://github.com/saltstack/salt/issues/56597). We only see this stack trace if the `gmp.dll` file is present on the system in a location in the path.

I uninstalled pycryptodomex and built pycryptodome from source following the instructions here:
https://www.pycryptodome.org/en/latest/src/installation.html#windows-from-sources-python-3-5-and-newer

I ran the test as in the instructions (`> python -m Cryptodome.SelfTest`) and noticed the following message:

```
Skipping GMP tests (ctypes.util.find_library() did not manage to locate a library called 'gmp')
```

So I dropped the dll ([gmp.zip](https://github.com/Legrandin/pycryptodome/files/4767143/gmp.zip)) in the C:\Windows directory and ran again... I now get the same error I'm seeing in Salt when I launch the SelfTest:

```
C:\dev\salt>python -m Cryptodome.SelfTest
Traceback (most recent call last):
  File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python37\lib\site-packages\Cryptodome\SelfTest\__main__.py", line 38, in <module>
    SelfTest.run(stream=sys.stdout, verbosity=1, config=config)
  File "C:\Python37\lib\site-packages\Cryptodome\SelfTest\__init__.py", line 61, in run
    tests = get_tests(config=config)
  File "C:\Python37\lib\site-packages\Cryptodome\SelfTest\__init__.py", line 90, in get_tests
    from Cryptodome.SelfTest import Math;   tests += Math.get_tests(config=config)
  File "C:\Python37\lib\site-packages\Cryptodome\SelfTest\Math\__init__.py", line 41, in get_tests
    tests += test_Numbers.get_tests(config=config)
  File "C:\Python37\lib\site-packages\Cryptodome\SelfTest\Math\test_Numbers.py", line 744, in get_tests
    from Cryptodome.Math._IntegerGMP import IntegerGMP
  File "C:\Python37\lib\site-packages\Cryptodome\Math\_IntegerGMP.py", line 95, in <module>
    lib = load_lib("gmp", gmp_defs)
  File "C:\Python37\lib\site-packages\Cryptodome\Util\_raw_api.py", line 102, in load_lib
    ffi.cdef(cdecl)
  File "C:\Python37\lib\site-packages\cffi\api.py", line 112, in cdef
    self._cdef(csource, override=override, packed=packed, pack=pack)
  File "C:\Python37\lib\site-packages\cffi\api.py", line 126, in _cdef
    self._parser.parse(csource, override=override, **options)
  File "C:\Python37\lib\site-packages\cffi\cparser.py", line 347, in parse
    self._internal_parse(csource)
  File "C:\Python37\lib\site-packages\cffi\cparser.py", line 387, in _internal_parse
    decl.type, name=decl.name, partial_length_ok=True)
  File "C:\Python37\lib\site-packages\cffi\cparser.py", line 619, in _get_type_and_quals
    tp = self._get_struct_union_enum_type('struct', type, name)
  File "C:\Python37\lib\site-packages\cffi\cparser.py", line 751, in _get_struct_union_enum_type
    self._declare('anonymous %s' % tp.forcename, tp)
  File "C:\Python37\lib\site-packages\cffi\cparser.py", line 526, in _declare
    "try cdef(xx, override=True))" % (name,))
cffi.FFIError: multiple declarations of anonymous MPZ (for interactive usage, try cdef(xx, override=True))
```

**Comment by twangboy** [comment]
Additionally, if I modify line 102 of `lib/Crypto/Util/_raw_api.py` with the recommended parameter like so:

```
ffi.cdef(cdecl, override=True)
```
I get a single warning when I run the tests:
```
C:\dev\salt>python -m Cryptodome.SelfTest
Skipping GMP tests (Not using GMP on Windows)
```
And then the tests continue successfully.

**Comment by twangboy** [comment]
I don't know if that is the correct fix, but I am submitting a PR.

**Comment by VeNoMouS** [comment]
I assume its `GNU Multiple Precision Arithmetic Library`

**Comment by Legrandin** [comment]
Thanks for reporting. I fixed this with f499fd3756f9f0a215bf4131992cfd03fb8b0a10 on master in a slightly different way, by skipping altogether loading of GMP when on Windows.

**Comment by twangboy** [comment]
Thanks for fixing that!
